### PR TITLE
[platform] [WIP] make setup constants configurable in gn

### PR DESCRIPTION
**Pull request by gjc13**
Now you can use:
```
extra_args="chip_device_test_setup_pincode=87654321 \
chip_device_test_setup_discriminator=1024" ./gn_build.sh
```
to specify the setup constants.

Also updated the device name in ble advertisements to make the devices more distinguishable.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]
Should there also be some documentation changes to accompany this change?

File: src/platform/BUILD.gn [hunk 1]
```diff
@@ -37,6 +37,12 @@ if (chip_device_platform != "none" && chip_device_platform != "external") {
 
     # Time the firmware was built.
     chip_device_config_firmware_build_time = ""
+
+    # Test setup pincode
+    chip_device_test_setup_pincode = ""
```
Comment (mspang) [inline src/platform/BUILD.gn:42]: Please use the same name for ifdef and GN arg.

**Review by mspang — CHANGES_REQUESTED** [review]
Can we pass this on the command line instead of environment?

**Comment by gjc13** [comment]
*DO NOT SUBMIT*

Have conflicts after https://github.com/project-chip/connectedhomeip/pull/2270 is merged.

We need to find a way to unify how we configure these macros.

**Review by woody-apple — APPROVED** [review]

File: gn_build.sh [hunk 2]
```diff
@@ -99,6 +99,25 @@ else
     echo "Hint: Set \$EFR32_SDK_ROOT to enable building for EFR32"
 fi
 
+opts=$(getopt --longoptions "pincode:,discriminator:" -n "$0" --options "" -- "$@")
```
Comment (mspang) [inline gn_build.sh:102]: Disagree with creating new bespoke options here, instead we could just generically accept args:

`gn_build.sh chip_device_config_use_test_setup_pin_code=12345`

But really, if you want a custom configuration, I'd actually recommend not using gn_build.sh at all and following the documented instructions for creating a custom configuration.

gn_build.sh is more intended to build what CI builds, without any local deviations.

**Review by mspang — CHANGES_REQUESTED** [review]

**Comment by rwalker-apple** [comment]
@gjc13 are you still working on this?

**Pull request closed without merging**